### ISSUE TITLE
Error when building automake from spec file

### ISSUE THREAD

**Issue by obreitwi** [spack/spack#12306]
While building `automake` the "regular" way (i.e., `spack install autoamke`) works, when building from a spec file, `autoconf` is missing.

### Steps to reproduce the issue

```console
$ spack spec -y automake %gcc@8.3.0 > automake.yaml
$ spack --verbose install -f automake.yaml
```

### Error Message

```console
==> Installing automake
==> Searching for binary cache of automake
==> Finding buildcaches in /wang/environment/software/jessie/spack/mirror/build_cache
==> Finding buildcaches in /opt/build_cache/build_cache
==> No binary for automake found: installing from source
==> Fetching file:///opt/build_cache/automake/automake-1.16.1.tar.gz
curl: (37) Couldn't open file /opt/build_cache/automake/automake-1.16.1.tar.gz
==> Fetching from file:///opt/build_cache/automake/automake-1.16.1.tar.gz failed.
==> Fetching file:///wang/environment/software/jessie/spack/mirror/automake/automake-1.16.1.tar.gz
######################################################################## 100.0%
==> Staging archive: /fasthome/obreitwi/sandboxes/2019-08-07_spack_from_file/spack/var/spack/stage/automake-1.16.1-xqnhkwm445wuehl53lbthk2qredk2kfj/automake-1.16.1.tar.gz                                                                                                                                                    
==> Created stage in /fasthome/obreitwi/sandboxes/2019-08-07_spack_from_file/spack/var/spack/stage/automake-1.16.1-xqnhkwm445wuehl53lbthk2qredk2kfj
==> Ran patch() for automake
==> Building automake [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> [2019-08-07-15:42:54.768883] '/fasthome/obreitwi/sandboxes/2019-08-07_spack_from_file/spack/var/spack/stage/automake-1.16.1-xqnhkwm445wuehl53lbthk2qredk2kfj/spack-src/configure' '--prefix=/fasthome/obreitwi/sandboxes/2019-08-07_spack_from_file/spack/opt/spack/linux-debian9-x86_64/gcc-8.3.0/automake-1.16.1-xqnhkwm445wuehl53lbthk2qredk2kfj'
checking whether make supports nested variables... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether ln -s works... yes
checking for perl... /fasthome/obreitwi/sandboxes/2019-08-07_spack_from_file/spack/opt/spack/linux-debian9-x86_64/gcc-8.3.0/perl-5.26.2-asacrqlh3dd6tgib5eps7xefe3zyj5zg/bin/perl                                                                                                                                             
checking for tex... tex
checking for yacc... no
checking for byacc... no
checking for bison... no
checking for lex... no
checking for flex... no
checking whether autoconf is installed... no
configure: error: Autoconf 2.65 or better is required.
    Please make sure it is installed and in your PATH.
==> Error: ProcessError: Command exited with status 1:
    '/fasthome/obreitwi/sandboxes/2019-08-07_spack_from_file/spack/var/spack/stage/automake-1.16.1-xqnhkwm445wuehl53lbthk2qredk2kfj/spack-src/configure' '--prefix=/fasthome/obreitwi/sandboxes/2019-08-07_spack_from_file/spack/opt/spack/linux-debian9-x86_64/gcc-8.3.0/automake-1.16.1-xqnhkwm445wuehl53lbthk2qredk2kfj'   

1 error found in build log:
     15    checking for yacc... no
     16    checking for byacc... no
     17    checking for bison... no
     18    checking for lex... no
     19    checking for flex... no
     20    checking whether autoconf is installed... no
  >> 21    configure: error: Autoconf 2.65 or better is required.
     22        Please make sure it is installed and in your PATH.
```

### Contents of the spec file: `automake.yaml`
```yaml
spec:
- automake:
    version: 1.16.1
    arch:
      platform: linux
      platform_os: debian9
      target: x86_64
    compiler:
      name: gcc
      version: 8.3.0
    namespace: builtin
    parameters:
      cflags: []
      cppflags: []
      cxxflags: []
      fflags: []
      ldflags: []
      ldlibs: []
    dependencies:
      perl:
        hash: asacrqlh3dd6tgib5eps7xefe3zyj5zg
        type:
        - build
        - run
    hash: xqnhkwm445wuehl53lbthk2qredk2kfj
- perl:
    version: 5.26.2
    arch:
      platform: linux
      platform_os: debian9
      target: x86_64
    compiler:
      name: gcc
      version: 8.3.0
    namespace: builtin
    parameters:
      cpanm: true
      patches:
      - 0eac10ed90aeb0459ad8851f88081d439a4e41978e586ec743069e8b059370ac
      shared: true
      threads: true
      cflags: []
      cppflags: []
      cxxflags: []
      fflags: []
      ldflags: []
      ldlibs: []
    patches:
    - 0eac10ed90aeb0459ad8851f88081d439a4e41978e586ec743069e8b059370ac
    dependencies:
      gdbm:
        hash: pdri3m3allxqpiwdikljeazpnuc5sevj
        type:
        - build
        - link
    hash: asacrqlh3dd6tgib5eps7xefe3zyj5zg
- gdbm:
    version: 1.18.1
    arch:
      platform: linux
      platform_os: debian9
      target: x86_64
    compiler:
      name: gcc
      version: 8.3.0
    namespace: builtin
    parameters:
      cflags: []
      cppflags: []
      cxxflags: []
      fflags: []
      ldflags: []
      ldlibs: []
    dependencies:
      readline:
        hash: bvf6tzqqhxktmj6ewh4lhnpqivcvneid
        type:
        - build
        - link
    hash: pdri3m3allxqpiwdikljeazpnuc5sevj
- readline:
    version: '7.0'
    arch:
      platform: linux
      platform_os: debian9
      target: x86_64
    compiler:
      name: gcc
      version: 8.3.0
    namespace: builtin
    parameters:
      cflags: []
      cppflags: []
      cxxflags: []
      fflags: []
      ldflags: []
      ldlibs: []
    dependencies:
      ncurses:
        hash: r6gwysm6ahzhqdnp7gt5ygdge6jjw3yw
        type:
        - build
        - link
    hash: bvf6tzqqhxktmj6ewh4lhnpqivcvneid
- ncurses:
    version: '6.1'
    arch:
      platform: linux
      platform_os: debian9
      target: x86_64
    compiler:
      name: gcc
      version: 8.3.0
    namespace: builtin
    parameters:
      symlinks: false
      termlib: false
      cflags: []
      cppflags: []
      cxxflags: []
      fflags: []
      ldflags: []
      ldlibs: []
    hash: r6gwysm6ahzhqdnp7gt5ygdge6jjw3yw
```

Afterwards, only the following packages are installed:
```console
$ spack find -dvL
==> 4 installed packages
-- linux-debian9-x86_64 / gcc@8.3.0 -----------------------------
pdri3m3allxqpiwdikljeazpnuc5sevj    gdbm@1.18.1
bvf6tzqqhxktmj6ewh4lhnpqivcvneid        ^readline@7.0
r6gwysm6ahzhqdnp7gt5ygdge6jjw3yw            ^ncurses@6.1~symlinks~termlib

r6gwysm6ahzhqdnp7gt5ygdge6jjw3yw    ncurses@6.1~symlinks~termlib

asacrqlh3dd6tgib5eps7xefe3zyj5zg    perl@5.26.2+cpanm patches=0eac10ed90aeb0459ad8851f88081d439a4e41978e586ec743069e8b059370ac +shared+threads
pdri3m3allxqpiwdikljeazpnuc5sevj        ^gdbm@1.18.1
bvf6tzqqhxktmj6ewh4lhnpqivcvneid            ^readline@7.0
r6gwysm6ahzhqdnp7gt5ygdge6jjw3yw                ^ncurses@6.1~symlinks~termlib

bvf6tzqqhxktmj6ewh4lhnpqivcvneid    readline@7.0
r6gwysm6ahzhqdnp7gt5ygdge6jjw3yw        ^ncurses@6.1~symlinks~termlib
```

I have not yet investigated as to why the `autoconf` binaries are not added to the environment in case the user specifies a spec yaml file.

**Comment by scheibelp** [comment]
Hi @obreitwi does https://github.com/spack/spack/pull/12317 resolve the issue for you?

**Comment by obreitwi** [comment]
> Hi @obreitwi does #12317 resolve the issue for you?

Hi, yes, the issue is resolved by #12317 - thank you very much!
